android/anc/src/main/java/org/smartregister/fhircore/anc/ui/madx/details/adapter/AllergiesAdapter.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.anc.ui.madx.details.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import org.smartregister.fhircore.anc.data.madx.model.AllergiesItem
+import org.smartregister.fhircore.anc.databinding.ItemPlanTextBinding
+
+/** Subclass of [ListAdapter] used to display allergies for the non ANC client */
+class AllergiesAdapter :
+  ListAdapter<AllergiesItem, AllergiesAdapter.PatientAllergiesViewHolder>(
+    ImmunizationItemDiffCallback
```

Comment:
Does this need to be specific to Immunization?

Reply:
Sorry for this, this code needs to be refactored. Every adapter has it's own DiffCallback.

Reply:
cool yea no worries :)

Reply:
code refactored.